src/modules/commander/Commander.cpp

```diff
@@ -1110,44 +1121,57 @@ Commander::handle_command(const vehicle_command_s &cmd)
 
 	case vehicle_command_s::VEHICLE_CMD_PREFLIGHT_REBOOT_SHUTDOWN: {
 
-			const int param1 = cmd.param1;
-
-			if (param1 == 0) {
+			switch (int(cmd.param1)) {
+			case 0:
 				// 0: Do nothing for autopilot
 				answer_command(cmd, vehicle_command_ack_s::VEHICLE_CMD_RESULT_ACCEPTED);
+				break;
 
 #if defined(CONFIG_BOARDCTL_RESET)
 
-			} else if ((param1 == 1) && shutdownIfAllowed() && (px4_reboot_request(false, 400_ms) == 0)) {
-				// 1: Reboot autopilot
-				answer_command(cmd, vehicle_command_ack_s::VEHICLE_CMD_RESULT_ACCEPTED);
+			case 1:
+				if (shutdownIfAllowed() && (px4_reboot_request(false, 400_ms) == 0)) {
+					// 1: Reboot autopilot
+					answer_command(cmd, vehicle_command_ack_s::VEHICLE_CMD_RESULT_ACCEPTED);
 
-				while (1) { px4_usleep(1); }
+					while (1) { px4_usleep(1); }
+				}
+
+				break;
 
 #endif // CONFIG_BOARDCTL_RESET
 
 #if defined(BOARD_HAS_POWER_CONTROL)
 
-			} else if ((param1 == 2) && shutdownIfAllowed() && (px4_shutdown_request(400_ms) == 0)) {
-				// 2: Shutdown autopilot
-				answer_command(cmd, vehicle_command_ack_s::VEHICLE_CMD_RESULT_ACCEPTED);
+			case 2:
+				if (shutdownIfAllowed() && (px4_shutdown_request(400_ms) == 0)) {
+					// 2: Shutdown autopilot
+					answer_command(cmd, vehicle_command_ack_s::VEHICLE_CMD_RESULT_ACCEPTED);
 
-				while (1) { px4_usleep(1); }
+					while (1) { px4_usleep(1); }
+				}
+
+				break;
 
 #endif // BOARD_HAS_POWER_CONTROL
 
 #if defined(CONFIG_BOARDCTL_RESET)
 
-			} else if ((param1 == 3) && shutdownIfAllowed() && (px4_reboot_request(true, 400_ms) == 0)) {
-				// 3: Reboot autopilot and keep it in the bootloader until upgraded.
-				answer_command(cmd, vehicle_command_ack_s::VEHICLE_CMD_RESULT_ACCEPTED);
+			case 3:
```

Comment:
This isn't the same. If `param1 == 3` but that `shutdownIfAllowed() && (px4_reboot_request(true, 400_ms) == 0)` is false, it will skip the `if` and exit the `switch`. Before, the code would go to the last `else` and send a `VEHICLE_CMD_RESULT_DENIED`.